frontend/src/mixins/suggestions-ctl.ts

```diff
@@ -96,29 +85,28 @@ export const SearchaliciousTermsMixin = <T extends Constructor<LitElement>>(
       q: string,
       taxonomyNames: string[]
     ): Promise<TaxomiesTermsResponse> {
-      this._setIsLoadingByTaxonomyNames(taxonomyNames, true);
+      this.isTermsLoading = true;
+      // get the version of the terms for each taxonomy
+      const version = this.incrementVersion();
+
       return fetch(this._termsUrl(q, taxonomyNames), {
         method: 'GET',
         headers: {
           'Content-Type': 'application/json',
         },
       })
         .then((response) => {
-          this._setIsLoadingByTaxonomyNames(taxonomyNames, false);
-
+          this.isTermsLoading = false;
```

Comment:
You need to guard it also, to avoid an abandoned query to modify this state.

```suggestion
          if (this.isLatestVersion(version)) {
            // this is the legitimate suggestion request, loading finished
            this.isTermsLoading = false;
          }
```